### PR TITLE
Add Application Insights alerts

### PR DESCRIPTION
**Pull request by ymehdimsft**
# Overview/Summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

Adds 2 alerts (App Insights Throttling limit reached, App Insights Delete) as per ADO work items [#35145](https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_sprints/backlog/Azure%20Landing%20Zones%20AMBA%20Team/Azure%20Landing%20Zones/Selenium/202412?workitem=35145) and [#36469](https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_sprints/backlog/Azure%20Landing%20Zones%20AMBA%20Team/Azure%20Landing%20Zones/Selenium/202412?workitem=36469)

**_Deployment results:_**

image

image

**_Remediation results:_**

image

image

image

### Breaking Changes

1. *NONE*

## As part of this Pull Request I have

- [x] Read the Contribution Guide and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/azure-monitor-baseline-alerts/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/azure-monitor-baseline-alerts/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/azure-monitor-baseline-alerts/)
- [x] Ensured PR tests are passing
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: patterns/alz/alzArm.param.json [hunk 1]
```diff
@@ -1660,6 +1660,45 @@
         },
         "LAWDailyCapLimitAlertState": {
           "value": "true"
+        },
+        "AppInsightsThrottlingLimitSeverity": {
```
Comment (arjenhuitema) [inline patterns/alz/alzArm.param.json:1664]: The new paramaters should go to the `policyAssignmentParametersWeb` section.

File: patterns/alz/policySetDefinitions/Deploy-Management-Alerts.json [hunk 2]
```diff
@@ -446,6 +446,163 @@
           "displayName": "ALog Analytics Workspace Daily Cap Limit Reached Alert State",
           "description": "Alert state for the alert"
         }
+      },
```
Comment (arjenhuitema) [inline patterns/alz/policySetDefinitions/Deploy-Management-Alerts.json:449]: Move the new policies references and paramaters to the `Deploy-Web-Alerts.json`

File: patterns/alz/templates/policies-Monitoring.bicep [hunk 3]
```diff
@@ -71,6 +71,8 @@ var loadPolicyDefinitions = {
     loadTextContent('../../../services/OperationalInsights/workspaces/Deploy-ActivityLog-LAWorkspace-Del.json')
     loadTextContent('../../../services/OperationalInsights/workspaces/Deploy-ActivityLog-LAWorkspace-KeyRegen.json')
     loadTextContent('../../../services/OperationalInsights/workspaces/Deploy-LAWorkspace-DailyCapLimitReached-Alert.json')
+    loadTextContent('../../../services/Insights/components/Deploy-AppInsightsThrottlingLimit-Alert.json')
```
Comment (arjenhuitema) [inline patterns/alz/templates/policies-Monitoring.bicep:74]: Remove the new lines from here and move them to `policies-Web.bicep`

File: patterns/alz/templates/policies.bicep [hunk 4]
```diff
@@ -167,6 +167,7 @@ var loadPolicyDefinitions = {
     loadTextContent('../../../services/OperationalInsights/workspaces/Deploy-ActivityLog-LAWorkspace-Del.json')
     loadTextContent('../../../services/OperationalInsights/workspaces/Deploy-ActivityLog-LAWorkspace-KeyRegen.json')
     loadTextContent('../../../services/OperationalInsights/workspaces/Deploy-LAWorkspace-DailyCapLimitReached-Alert.json')
+    loadTextContent('../../../services/Insights/components/Deploy-AppInsightsThrottlingLimit-Alert')
```
Comment (arjenhuitema) [inline patterns/alz/templates/policies.bicep:170]: Please undo this change. This file is no longer required.

**Review by arjenhuitema — CHANGES_REQUESTED** [review]
Thanks @ymehdimsft 

Let's work on making the following changes:
1. These resource types typically dont exist under the `Management` management group. Therefore we should include these policies not in the Management Initiative. As this regards App Insights it makes sense to add them to the Web Initiative.
2. The `policies.bicep` will no longer be used, and shouldn't be modified anymore and it will be removed in the near future.

**Review by ymehdimsft — COMMENTED** [review]
All comments adressed

**Review by Brunoga-MS — APPROVED** [review]
LGTM

**Review by Brunoga-MS — APPROVED** [review]
LGTM
